src/attacks.rs

```diff
@@ -78,6 +81,31 @@ fn greedy_reduce(g: &mut Graph, target: usize, p: GreedyParams) -> HashSet<usize
     s
 }
 
+// Variation of Greedy attack that has as `target` the resulting size of set S.
+// FIXME: To hit exactly the `target_size` we should consider the number of nodes
+//  removed in each iteration (`GreedyParams::k`), but since that number is small
+//  compared to normal target sizes it is an acceptable bias for now.
+fn greedy_reduce_by_size(g: &mut Graph, target_size: usize, p: GreedyParams) -> HashSet<usize> {
```

Comment:
Duplicated from `greedy_reduce_to_depth`. Abstract in a similar way to `valiant_reduce`.